src/pudl/scripts/dbt_helper.py

```diff
@@ -281,13 +284,18 @@ def _calculate_row_counts(
 
     row_count_query = f"""
 SELECT
-    CAST({partition_expr_sql} AS VARCHAR) AS partition,
+    CAST(COALESCE(CAST({partition_expr_sql} AS VARCHAR), '') AS VARCHAR) AS partition,
```

Comment:
Unfortunately this change along wasn't enough to make the old and new dataframes comparable. I moved the additional munging up into the functions that read the old/new row counts though, rather than having it in the update function.